stdlib/public/core/Integers.swift

```diff
@@ -1694,23 +1711,108 @@ extension BinaryInteger {
       // undefined behavior to access the excess allocation.
       let buffer =
         unsafe UnsafeMutableBufferPointer<UTF8.CodeUnit>(
-          start: $0.baseAddress, count: capacity)
+          start: $0.baseAddress,
+          count: capacity)
       unsafe buffer.initialize(repeating: 0x30)
       defer { unsafe buffer.deinitialize() }
 
-      var span = unsafe buffer.mutableSpan
-      let textRange = unsafe _BinaryIntegerToASCII(
+      // Work in chunks for speed.
```

Comment:
This is the beginning of the substantive part of this PR. As before, it's been tested thoroughly in a separate package.

I've tried to be very careful of the following, which is not so testable (at least with my setup), but it could use some eyes during review:
* Any endianness issues with going chunk-by-chunk in the specific way that I do (I don't think so, but I've been fooled in the past).
* Whether the strategy described above (see line R1695) with `safetyMargin` is indeed capable of avoiding buffer overflows even with hypothetical end-user `BinaryInteger` types that may be unusually sized or even incorrectly implemented.